src/components/rule-builder/input-selection-box.component.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState, useRef, useCallback, useMemo } from 'react';
+import { TextInput, MultiSelect } from '@carbon/react';
+import { useDebounce } from '@openmrs/esm-framework';
+import { useTranslation } from 'react-i18next';
+import styles from './input-selection-box.scss';
+
+interface CustomComboBoxProps {
+  id: string;
+  items: Array<Record<string, string>>;
```

Comment:
Shouldn't the type here be a little more specific? I.e., we require a `label` property for this to work and it seems like we also expect a `concept` property... and ignore anything else.